mongo/mongo_test.go

```diff
@@ -339,7 +397,36 @@ func (s *MongoSuite) TestInstallMongodServiceExists(c *gc.C) {
 
 	// We still attempt to install mongodb, despite the service existing.
 	cmds := getMockShellCalls(c, output)
-	c.Check(cmds, gc.HasLen, 1)
+
+	if s.PackageManager == "yum" {
+		c.Assert(cmds, gc.DeepEquals, [][]string{
+			{"--assumeyes", "--debuglevel=1", "install", "epel-release"},
+			{"--assumeyes", "--debuglevel=1", "install", "mongodb-server"},
+		})
+		
+		c.Assert(getMockShellCalls(c, chconOut), gc.DeepEquals, [][]string{{
+			"-R",
+			"-v",
+			"-t",
+			"mongod_var_lib_t",
+			"/var/lib/juju/",
+		}})
+
+		c.Assert(getMockShellCalls(c, semanageOut), gc.DeepEquals, [][]string{{
+			"port",
+			"-a",
+			"-t",
+			"mongod_port_t",
+			"-p",
+			"tcp",
+			"37017",
+		}})
+	} else {
+		c.Check(cmds, gc.HasLen, 1)
```

Comment:
This ends up looking really weird. 
Either test this version of cmd's with deepequals, or test that you have above with length(I'd prefer deep equals).

On a second thought, these might be better off as split tests.